stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/chat/info/group/GroupChatInfoFragment.kt

```diff
@@ -0,0 +1,112 @@
+
+    private var _binding: FragmentGroupChatInfoBinding? = null
+    private val binding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentGroupChatInfoBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.optionsRecyclerView.adapter = adapter
+        headerViewModel.bindView(binding.headerView, viewLifecycleOwner)
+        bindGroupInfoViewModel()
+    }
+
+    override fun onDestroyView() {
+        _binding = null
+        super.onDestroyView()
+    }
+
+    private fun bindGroupInfoViewModel() {
+        subscribeForChannelMutesUpdatedEvents()
+        setOnClickListeners()
+
+        viewModel.channelLeftState.observe(viewLifecycleOwner) { wasLeft ->
+            if (wasLeft) {
+                findNavController().popBackStack(R.id.homeFragment, false)
+            }
+        }
+
+        viewModel.state.observe(viewLifecycleOwner) { state ->
+            val members = if (state.shouldExpandMembers != false) {
+                state.members.map { ChatInfoItem.MemberItem(it) }
+            } else {
+                state.members.take(GroupChatInfoViewModel.COLLAPSED_MEMBERS_COUNT)
+                    .map { ChatInfoItem.MemberItem(it) } + ChatInfoItem.MembersSeparator(state.membersToShowCount)
+            }
+            adapter.submitList(
+                members +
+                    listOf(
+                        ChatInfoItem.Separator,
+                        ChatInfoItem.ChannelName(state.channelName),
+                        ChatInfoItem.Option.Stateful.MuteChannel(isChecked = state.channelMuted),
+                        ChatInfoItem.Option.SharedMedia,
+                        ChatInfoItem.Option.SharedFiles,
+                        ChatInfoItem.Option.LeaveGroup,
+                    )
+            )
+        }
+    }
+
+    private fun setOnClickListeners() {
+        adapter.setChatInfoStatefulOptionChangedListener { option, isChecked ->
+            viewModel.onAction(
+                when (option) {
+                    is ChatInfoItem.Option.Stateful.MuteChannel -> GroupChatInfoViewModel.Action.MuteChannelClicked(
+                        isChecked
+                    )
+                    else -> throw IllegalStateException("Chat info option $option is not supported!")
+                }
+            )
+        }
+        adapter.setChatInfoOptionClickListener { option ->
+            when (option) {
+                ChatInfoItem.Option.SharedMedia -> Unit // TODO: Not supported yet
+                ChatInfoItem.Option.SharedFiles -> Unit // TODO: Not supported yet
+                ChatInfoItem.Option.LeaveGroup -> viewModel.onAction(GroupChatInfoViewModel.Action.LeaveChannelClicked)
```

Comment:
TODO: some kind of confirmation dialog before leaving, I was checking out the screen and accidentally left a group 😄

Reply:
I'll add a task for that